features/step_definitions/data.js

```diff
@@ -48,41 +48,26 @@ module.exports = function () {
         q.awaitAll(callback);
     });
 
-    this.Given(/^the node map$/, (table, callback) => {
-        let q = d3.queue();
-
-        let addNode = (name, ri, ci, cb) => {
-            if (name) {
-                let nodeWithID = name.match(/([a-z])\:([0-9]*)/);
-                if (nodeWithID) {
-                    let nodeName = nodeWithID[1],
-                        nodeID = nodeWithID[2];
-                    if (this.nameNodeHash[nodeName]) throw new Error(util.format('*** duplicate node %s', name));
-                    let lonLat = this.tableCoordToLonLat(ci, ri);
-                    this.addOSMNode(nodeName, lonLat[0], lonLat[1], nodeID);
-                } else {
-                    if (name.length !== 1) throw new Error(util.format('*** node invalid name %s, must be single characters', name));
-                    if (!name.match(/[a-z0-9]/)) throw new Error(util.format('*** invalid node name %s, must me alphanumeric', name));
-
-                    if (name.match(/[a-z]/)) {
-                        if (this.nameNodeHash[name]) throw new Error(util.format('*** duplicate node %s', name));
-                        let lonLat = this.tableCoordToLonLat(ci, ri);
-                        this.addOSMNode(name, lonLat[0], lonLat[1], null);
-                    } else {
-                        if (this.locationHash[name]) throw new Error(util.format('*** duplicate node %s'), name);
-                        let lonLat = this.tableCoordToLonLat(ci, ri);
-                        this.addLocation(name, lonLat[0], lonLat[1], null);
-                    }
-                }
-
-                cb();
+    this.Given(/^the node map$/, (docstring, callback) => {
+        var q = d3.queue();
+
+        var addNode = (name, ri, ci, cb) => {
+            var lonLat = this.tableCoordToLonLat(ci, ri);
+            if (name.match(/[a-z]/) ) {
+                if (this.nameNodeHash[name]) throw new Error(util.format('*** duplicate node %s', name));
+                this.addOSMNode(name, lonLat[0], lonLat[1], null);
```

Comment:
Previously, id could be set in the node map using the special syntax 'a:14'. This doesn't work with doc strings, since you cannot insert extra characters.
I changed it so you can specify IDs using the existing 'And the nodes' tables, using an 'id' column. There are a few tests that uses this.